federation/normalize.go

```diff
@@ -150,8 +150,14 @@ func mergeSameAlias(selections []*graphql.Selection) ([]*graphql.Selection, erro
 		if last == nil || selection.Alias != last.Alias {
 			// Make a copy of the selection so we can modify it below
 			// or when we flatten recursively later.
-			copy := *selection
-			selection = &copy
+			cp := *selection
+			if selection.SelectionSet != nil {
+				// Make a new SelectionSet for the copy so we will not append to the original slice later.
+				cp.SelectionSet = &graphql.SelectionSet{}
+				cp.SelectionSet.Selections = append(cp.SelectionSet.Selections, selection.SelectionSet.Selections...)
+				cp.SelectionSet.Fragments = append(cp.SelectionSet.Fragments, selection.SelectionSet.Fragments...)
```

Comment:
Maintainability: if we add a new field to SelectionSet, we'll need to remember to update this call site. What do you think of adding a method to SelectionSet that returns a shallow copy, instead, to try to make it easier to do that if we do add a field later?